src/core/trace_core.mli

```diff
@@ -79,10 +80,9 @@ val exit_manual_span : explicit_span -> unit
     {!enter_manual_toplevel_span}.
     @since 0.3 *)
 
-val add_data_to_explicit_span :
-  explicit_span -> (string * user_data) list -> unit
-(** [add_data_explicit esp data] is [add_data_to_span esp.span data], ie. it adds
-    the pairs [k:v] from [data] to the span inside [esp].
+val add_data_to_manual_span : explicit_span -> (string * user_data) list -> unit
+(** [add_data_explicit esp data] adds [data] to the span [esp].
+      The behavior is not specified is the span has been exited already.
```

Comment:
Ditto the above, except that we can't really save the data and add it to a span later; so the available options that we should choose from and strongly specify are №s.2 and 3 …

Reply:
Imho adding the data to another span is also a bug. Why would metadata show on an unrelated span? That's confusing.